lib/cylc/registration.py

```diff
@@ -151,8 +154,23 @@ def get_list(self, regfilter=None):
     def unregister(self, exp):
         """Un-register a suite."""
         suitedirs = []
+
+        # get list of running suites
+        try:
+            ret = subprocess.Popen(['cylc', 'scan'], stdout=subprocess.PIPE)
```

Comment:
As discussed, this would not work well as `cylc scan` is a very expensive command to run.